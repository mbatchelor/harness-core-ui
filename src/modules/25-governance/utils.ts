import type { GovernanceMetadata } from 'services/cd-ng'

interface DoesGovernanceMetaDataHasErrorOrWarningRtn {
  governanceMetaDataHasError: boolean
  governanceMetaDataHasWarning: boolean
}
export const doesGovernanceHasErrorOrWarning = (
  governanceMetadata: GovernanceMetadata | undefined
): DoesGovernanceMetaDataHasErrorOrWarningRtn => {
  let governanceMetaDataHasWarning = false
  let governanceMetaDataHasError = false
  if (governanceMetadata && governanceMetadata?.status === 'error') {
    governanceMetaDataHasError = true
  }
  if (governanceMetadata && governanceMetadata?.status === 'warning') {
    governanceMetaDataHasWarning = true
  }
  return { governanceMetaDataHasError, governanceMetaDataHasWarning }
}
