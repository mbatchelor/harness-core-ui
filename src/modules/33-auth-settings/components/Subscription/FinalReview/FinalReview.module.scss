.line {
  border-bottom: 1px solid var(--grey-100);
  padding: var(--spacing-medium) 0 !important;
}

.payingToday {
  background-color: var(--primary-1) !important;
  border-radius: 4px;
  padding: var(--spacing-medium) !important;
}

.lineItem {
  width: 33%;
}
