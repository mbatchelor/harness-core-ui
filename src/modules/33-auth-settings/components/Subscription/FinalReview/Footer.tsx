/*
 * Copyright 2022 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

import React from 'react'
import { Layout, Button, ButtonVariation } from '@harness/uicore'
import { useStrings } from 'framework/strings'
import { SubscribeViews } from '@common/constants/SubscriptionTypes'
import type { TIME_TYPE } from '@auth-settings/pages/subscriptions/plans/planUtils'
import css from '@auth-settings/modals/Subscription/useSubscriptionModal.module.scss'

interface FooterProps {
  time?: TIME_TYPE
  setView: (view: SubscribeViews) => void
}

export const Footer = ({ setView }: FooterProps): React.ReactElement => {
  const { getString } = useStrings()

  function handleNext(): void {
    setView(SubscribeViews.SUCCESS)
  }

  function handleBack(): void {
    setView(SubscribeViews.BILLINGINFO)
  }

  return (
    <Layout.Horizontal className={css.footer}>
      <Layout.Horizontal spacing={'large'}>
        <Button variation={ButtonVariation.SECONDARY} onClick={handleBack} icon={'chevron-left'}>
          {getString('back')}
        </Button>
        <Button variation={ButtonVariation.PRIMARY} onClick={handleNext}>
          {getString('authSettings.billing.subscribeNPay')}
        </Button>
      </Layout.Horizontal>
    </Layout.Horizontal>
  )
}
