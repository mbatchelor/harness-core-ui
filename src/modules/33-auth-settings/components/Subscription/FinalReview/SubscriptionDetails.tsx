/*
 * Copyright 2022 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

import React from 'react'
import { Layout, Text } from '@harness/uicore'
import { Color, FontVariation } from '@harness/design-system'
import { useStrings } from 'framework/strings'
import type { Editions } from '@common/constants/SubscriptionTypes'
import css from '../CostCalculator/CostCalculator.module.scss'
import detailCss from './FinalReview.module.scss'

const Line = ({
  description,
  breakdown,
  price = 0
}: {
  description: string
  breakdown?: React.ReactElement
  price?: number
}): React.ReactElement => {
  return (
    <Layout.Horizontal flex={{ alignItems: 'start' }} className={detailCss.line}>
      <Text font={{ weight: 'semi-bold' }} className={detailCss.lineItem}>
        {description}
      </Text>
      <div className={detailCss.lineItem}> {breakdown}</div>
      <Text font={{ weight: 'semi-bold' }} padding={{ right: 'xlarge' }}>
        {price.toLocaleString('en-US', {
          style: 'currency',
          currency: 'USD'
        })}
      </Text>
    </Layout.Horizontal>
  )
}

const PayingToday = ({ total = 0 }: { total?: number }): React.ReactElement => {
  const { getString } = useStrings()
  return (
    <Layout.Horizontal className={detailCss.payingToday} flex={{ justifyContent: 'space-between' }}>
      <Text font={{ variation: FontVariation.H4 }}>{getString('authSettings.costCalculator.payingToday')}</Text>
      <Text font={{ variation: FontVariation.H4 }}>
        {total.toLocaleString('en-US', {
          style: 'currency',
          currency: 'USD'
        })}
      </Text>
    </Layout.Horizontal>
  )
}
export const SubscriptionDetails = ({ plan }: { plan: Editions }): React.ReactElement => {
  const { getString } = useStrings()
  const licenseBreakdown = (
    <Text>
      {getString('authSettings.unitPrice')}
      {getString('common.perMonth')}
    </Text>
  )
  const mausBreakdown = (
    <Layout.Vertical>
      <Text>
        {getString('authSettings.unitPrice')}
        {getString('common.perMonth')}
      </Text>
      <Text>{getString('authSettings.firstIncludedFree')}</Text>
    </Layout.Vertical>
  )
  const supportBreakdown = (
    <Layout.Horizontal>
      <Text>{getString('authSettings.finalReview.premiumSupport')}</Text>
      <Text>{getString('authSettings.finalReview.ofTotalSpend')}</Text>
    </Layout.Horizontal>
  )
  const autoRenew = <Text>{getString('authSettings.costCalculator.autoRenew')}</Text>

  return (
    <Layout.Vertical>
      <Layout.Horizontal spacing={'small'}>
        <Text font={{ variation: FontVariation.H5 }}>{getString('authSettings.costCalculator.newSubscription')}</Text>
        <Text
          font={{ weight: 'bold', size: 'xsmall' }}
          color={Color.WHITE}
          border={{ radius: 8 }}
          className={css.newPlan}
        >
          {plan}
        </Text>
      </Layout.Horizontal>
      <Layout.Vertical spacing={'large'}>
        <Line description={getString('authSettings.costCalculator.developerLicenses')} breakdown={licenseBreakdown} />
        <Line description={getString('authSettings.costCalculator.maus')} breakdown={mausBreakdown} />
        <Line description={getString('authSettings.costCalculator.support')} breakdown={supportBreakdown} />
        <Line description={getString('authSettings.costCalculator.tax')} />
        <Line description={getString('authSettings.yearlySubscriptionTotal')} breakdown={autoRenew} />
      </Layout.Vertical>
      <PayingToday />
    </Layout.Vertical>
  )
}
