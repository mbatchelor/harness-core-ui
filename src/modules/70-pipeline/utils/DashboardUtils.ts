/*
 * Copyright 2022 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

import type { TooltipFormatterContextObject } from 'highcharts'

interface PointStats {
  deployments?: {
    failure?: number
    success?: number
    total?: number
  }
  time?: number
}

/* istanbul ignore next */
export const renderTooltipContent = ({
  time,
  failureRate,
  count,
  successCount,
  failureCount
}: {
  time: string | number
  failureRate: string | number
  count?: number
  successCount?: number
  failureCount?: number
}) => {
  return `<div style="padding: 16px; color: white; width: 282px; height: 128px;">
        <div style="display: flex; justify-content: space-between; border-bottom: 0.5px solid rgba(243, 243, 250); padding-bottom: 7px; margin-bottom: 15px;">
          <div style="font-weight: normal; font-size: 12px; line-height: 18px; opacity: 0.8;">${time}</div>
          <div>
            <span style="white-space: pre; font-weight: bold; font-size: 12px; line-height: 18px; opacity: 0.8;">Deployments: </span>
            <span style="font-weight: bold; font-size: 12px; line-height: 18px;">${count}</span>
          </div>
        </div>
        <div style="display: flex; justify-content: space-between;">
          <div>
            <p style="font-weight: 500; font-size: 10px; line-height: 14px; letter-spacing: 0.2px; color: #D9DAE6; margin-bottom: 0px;">Failure Rate</p>
            <p style="font-weight: 600; font-size: 28px; line-height: 38px; color: #FBE6E4;">${failureRate}</p>
          </div>
          <div style="margin-right: 8px;">
            <div style="display: flex; align-items: center; margin-bottom: 6px;">
              <div style="height: 6px; width: 12px; background-color: #5FB34E; border-radius: 16px; display: inline-block; margin-right: 8px"></div>
              <span style="white-space: pre; font-weight: bold; font-size: 12px; line-height: 16px; letter-spacing: 0.2px; opacity: 0.8;">Success </span>
              <span style="font-weight: bold; font-size: 12px; line-height: 16px; letter-spacing: 0.2px;">${successCount}</span>
            </div>
            <div style="display: flex; align-items: center;">
              <div style="height: 6px; width: 12px; background-color: #EE5F54; border-radius: 16px; display: inline-block; margin-right: 8px"></div>
              <span style="white-space: pre; font-weight: bold; font-size: 12px; line-height: 18px; opacity: 0.8;">Failed </span>
              <span style="font-weight: bold; font-size: 12px; line-height: 18px;">${failureCount}</span>
            </div>
          </div>
        </div>
      </div>`
}

/* istanbul ignore next */
export const getTooltip = (currPoint: TooltipFormatterContextObject): string => {
  const custom = currPoint?.series?.userOptions?.custom
  const point: PointStats = custom?.[currPoint.key]
  const time =
    point && point?.time
      ? new Date(point?.time).toLocaleDateString('en-US', { day: 'numeric', month: 'long' })
      : currPoint.x
  let failureRate: string | number = 'Infinity'
  if (point?.deployments?.failure && point.deployments?.total) {
    failureRate = ((point.deployments.failure / point.deployments.total) * 100).toFixed(1) + '%'
  }
  if (point?.deployments?.failure === 0) {
    failureRate = '0'
  }
  return renderTooltipContent({
    time,
    failureRate,
    count: point?.deployments?.total,
    successCount: point?.deployments?.success,
    failureCount: point?.deployments?.failure
  })
}
