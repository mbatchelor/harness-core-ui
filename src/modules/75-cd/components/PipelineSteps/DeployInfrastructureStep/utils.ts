/*
 * Copyright 2022 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

import type { FormikProps } from 'formik'
import { isEmpty } from 'lodash-es'

import { getMultiTypeFromValue, MultiTypeInputType } from '@harness/uicore'
import type { EnvironmentResponseDTO, PipelineInfrastructure } from 'services/cd-ng'

import type { StepViewType } from '@pipeline/components/AbstractSteps/Step'

export interface DeployInfrastructureProps {
  initialValues: PipelineInfrastructure
  onUpdate?: (data: PipelineInfrastructure) => void
  readonly: boolean
  allowableTypes: MultiTypeInputType[]
  stepViewType?: StepViewType
  inputSetData?: {
    template?: PipelineInfrastructure
    path?: string
    readonly?: boolean
  }
}

export interface DeployInfrastructureState {
  isEdit: boolean
  isEnvironment: boolean
  formik?: FormikProps<PipelineInfrastructure>
  data?: EnvironmentResponseDTO
}

export function isEditEnvironment(data: PipelineInfrastructure): boolean {
  if (getMultiTypeFromValue(data.environmentRef) !== MultiTypeInputType.RUNTIME && !isEmpty(data.environmentRef)) {
    return true
  } else if (data.environment && !isEmpty(data.environment.identifier)) {
    return true
  }
  return false
}

export function isEditInfrastructure(data: any): boolean {
  if (
    getMultiTypeFromValue(data.infrastructureRef) !== MultiTypeInputType.RUNTIME &&
    !isEmpty(data.infrastructureRef)
  ) {
    return true
  } else if (data.infrastructure && !isEmpty(data.infrastructure.identifier)) {
    return true
  }
  return false
}
