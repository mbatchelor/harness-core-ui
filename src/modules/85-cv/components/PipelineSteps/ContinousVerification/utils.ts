/*
 * Copyright 2021 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

import { getMultiTypeFromValue, MultiTypeInputType, RUNTIME_INPUT_VALUE, SelectOption } from '@wings-software/uicore'
import { isEmpty, isNull, isUndefined, omit, omitBy, set } from 'lodash-es'
import { FormikErrors, yupToFormErrors } from 'formik'
import * as Yup from 'yup'
import type { UseStringsReturn } from 'framework/strings'
import { getDurationValidationSchema } from '@common/components/MultiTypeDuration/MultiTypeDuration'
import type { StringsMap } from 'stringTypes'
import type { ContinousVerificationData, spec, VerifyStepMonitoredService } from './types'
import {
  VerificationSensitivityOptions,
  durationOptions,
  baseLineOptions,
  trafficSplitPercentageOptions,
  SensitivityTypes,
  defaultMonitoredServiceSpec,
  monitoredServiceRefPath
} from './constants'
import { MONITORED_SERVICE_TYPE } from './components/ContinousVerificationWidget/components/ContinousVerificationWidgetSections/components/SelectMonitoredServiceType/SelectMonitoredServiceType.constants'

/**
 * checks if a field is a runtime input.
 * @param field
 * @returns boolean
 */
export function checkIfRunTimeInput(field: string | SelectOption | number | undefined): boolean {
  return getMultiTypeFromValue(field as string) === MultiTypeInputType.RUNTIME
}

/**
 * validates a particular field for errors.
 * @param fieldValue
 * @param fieldKey
 * @param data
 * @param errors
 * @param getString
 * @param isRequired
 */
export function validateField(
  fieldValue: string,
  fieldKey: string,
  data: ContinousVerificationData,
  errors: any,
  getString: UseStringsReturn['getString'],
  isRequired = true
): void {
  if (checkIfRunTimeInput(fieldValue) && isRequired && isEmpty(data?.spec?.spec && data?.spec?.spec[fieldKey])) {
    set(errors, `spec.spec.${fieldKey}`, getString('fieldRequired', { field: fieldKey }))
  }
}

export function validateMonitoredService(
  data: ContinousVerificationData,
  errors: FormikErrors<ContinousVerificationData>,
  getString: (key: keyof StringsMap, vars?: Record<string, any> | undefined) => string,
  isRequired: boolean,
  monitoredService?: VerifyStepMonitoredService
): void {
  if (
    checkIfRunTimeInput(monitoredService?.spec?.monitoredServiceRef) &&
    isRequired &&
    isEmpty(data?.spec?.monitoredService?.spec?.monitoredServiceRef)
  ) {
    set(errors, monitoredServiceRefPath, getString('fieldRequired', { field: 'Monitored service' }))
  }
}

/**
 * validates timeout field.
 * @param template
 * @param getString
 * @param data
 * @param errors
 */
export function validateTimeout(
  getString: UseStringsReturn['getString'],
  data: ContinousVerificationData,
  errors: any,
  template?: ContinousVerificationData,
  isRequired = true
): void {
  if (checkIfRunTimeInput(template?.timeout)) {
    let timeout: Yup.ObjectSchema
    if (isRequired) {
      timeout = Yup.object().shape({
        timeout: getDurationValidationSchema({ minimum: '10s' }).required(getString?.('validation.timeout10SecMinimum'))
      })
    } else {
      timeout = Yup.object().shape({
        timeout: getDurationValidationSchema({ minimum: '10s' })
      })
    }

    try {
      timeout.validateSync(data)
    } catch (e) {
      if (e instanceof Yup.ValidationError) {
        const err = yupToFormErrors(e)
        Object.assign(errors, err)
      }
    }
  }
}

/**
 * returns yaml data for spec field.
 * @param specInfo
 * @returns spec
 */
export function getSpecYamlData(specInfo?: spec, type?: string): spec {
  let validspec = omitBy(specInfo, v => isUndefined(v) || isNull(v) || v === '')
  switch (type) {
    case 'LoadTest':
      validspec = omit(validspec, ['trafficsplit'])
      break
    case 'Bluegreen':
    case 'Canary':
      validspec = omit(validspec, ['baseline'])
      break
    default:
      break
  }

  Object.keys(validspec).map((key: string) => {
    //TODO logic in if block will be removed once backend api is fixed : https://harness.atlassian.net/browse/CVNG-2481
    if (key === 'sensitivity') {
      validspec[key] = validspec[key].value
        ? SensitivityTypes[validspec[key].value as keyof typeof SensitivityTypes]
        : validspec[key]
    } else {
      validspec[key] = validspec[key].value ? validspec[key].value : validspec[key]
    }
  })

  return validspec
}

export function getMonitoredServiceYamlData(spec: ContinousVerificationData['spec']): VerifyStepMonitoredService {
  let monitoredService: VerifyStepMonitoredService = defaultMonitoredServiceSpec

  switch (spec?.monitoredService?.type) {
    case MONITORED_SERVICE_TYPE.DEFAULT:
      monitoredService = defaultMonitoredServiceSpec
      break
    case MONITORED_SERVICE_TYPE.CONFIGURED:
      monitoredService = {
        type: MONITORED_SERVICE_TYPE.CONFIGURED,
        spec: {
          monitoredServiceRef: getMonitoredServiceRef(spec)
        }
      }
      break
    default:
      monitoredService = defaultMonitoredServiceSpec
  }
  return monitoredService
}

export function getMonitoredServiceRef(spec: ContinousVerificationData['spec']): string {
  return spec?.monitoredService?.spec?.monitoredServiceRef as string
}

/**
 * returns forms data for spec field.
 * @param specInfo
 * @returns spec
 */
export function getSpecFormData(specInfo: spec | undefined): spec {
  const validspec: spec | undefined = { ...specInfo }
  if (specInfo) {
    Object.keys(specInfo).map((key: string) => {
      switch (key) {
        case 'sensitivity':
          setFieldData(validspec, 'sensitivity', VerificationSensitivityOptions)
          break
        case 'duration':
          setFieldData(validspec, 'duration', durationOptions)
          break
        case 'baseline':
          setFieldData(validspec, 'baseline', baseLineOptions)
          break
        case 'trafficsplit':
          setFieldData(validspec, 'trafficsplit', trafficSplitPercentageOptions)
          break
        default:
      }
    })
  }
  return validspec
}

/**
 * sets particular field data
 * @param validspec
 * @param field
 * @param options
 */
export function setFieldData(validspec: spec | undefined, field: string, fieldOptions: SelectOption[]): void {
  //finding the complete option if the field is fixed input
  if (validspec && validspec[field] && validspec[field] !== RUNTIME_INPUT_VALUE) {
    //TODO logic in if block will be removed once backend api is fixed : https://harness.atlassian.net/browse/CVNG-2481
    if (field === 'sensitivity') {
      validspec[field] = fieldOptions.find(
        (el: SelectOption) =>
          SensitivityTypes[el.value as keyof typeof SensitivityTypes] === (validspec && validspec[field])
      )
    } else {
      validspec[field] = fieldOptions.find((el: SelectOption) => el.value === (validspec && validspec[field]))
    }
  }
}
