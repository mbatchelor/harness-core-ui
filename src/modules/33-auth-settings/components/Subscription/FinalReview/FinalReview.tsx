/*
 * Copyright 2022 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

import React from 'react'
import { Layout } from '@harness/uicore'
import type { Module } from 'framework/types/ModuleName'
import type { Editions, SubscribeViews } from '@common/constants/SubscriptionTypes'
import { useStrings } from 'framework/strings'
import { Footer } from './Footer'
import { Header } from '../Header'
import { SubscriptionDetails } from './SubscriptionDetails'

interface FinalReviewProps {
  module: Module
  setView: (view: SubscribeViews) => void
  plan: Editions
}
export const FinalReview = ({ module, setView, plan }: FinalReviewProps): React.ReactElement => {
  const { getString } = useStrings()
  return (
    <Layout.Vertical>
      <Header module={module} stepDescription={getString('authSettings.finalReview.step')} step={3} />
      <SubscriptionDetails plan={plan} />
      <Footer setView={setView} />
    </Layout.Vertical>
  )
}
