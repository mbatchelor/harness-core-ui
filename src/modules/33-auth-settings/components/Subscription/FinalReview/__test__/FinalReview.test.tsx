/*
 * Copyright 2022 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

import React from 'react'
import { render, act, waitFor, fireEvent } from '@testing-library/react'
import { TestWrapper } from '@common/utils/testUtils'
import { Editions, SubscribeViews } from '@common/constants/SubscriptionTypes'
import { FinalReview } from '../FinalReview'

describe('FinalReview', () => {
  test('render', () => {
    const { container, getByText } = render(
      <TestWrapper>
        <FinalReview module="cf" setView={jest.fn()} plan={Editions.ENTERPRISE} />
      </TestWrapper>
    )
    expect(getByText('authSettings.finalReview.step')).toBeInTheDocument()
    expect(container).toMatchSnapshot()
  })

  test('footer', async () => {
    const setViewMock = jest.fn()
    const { getByText } = render(
      <TestWrapper>
        <FinalReview module="cf" setView={setViewMock} plan={Editions.TEAM} />
      </TestWrapper>
    )
    act(() => {
      fireEvent.click(getByText('back'))
    })
    await waitFor(() => {
      expect(setViewMock).toBeCalledWith(SubscribeViews.BILLINGINFO)
    })
    act(() => {
      fireEvent.click(getByText('authSettings.billing.subscribeNPay'))
    })
    await waitFor(() => {
      expect(setViewMock).toBeCalledWith(SubscribeViews.SUCCESS)
    })
  })
})
