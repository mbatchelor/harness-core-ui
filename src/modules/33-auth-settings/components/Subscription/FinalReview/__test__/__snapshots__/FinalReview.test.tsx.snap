// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FinalReview render 1`] = `
<div>
  <div
    class="Layout--vertical StyledProps--main"
  >
    <div
      class="Layout--vertical StyledProps--main StyledProps--padding-bottom-large"
    >
      <p
        class="StyledProps--font StyledProps--main StyledProps--font-variation-h4 StyledProps--padding-bottom-large"
        style="display: flex; align-items: center;"
      >
        <span
          class="bp3-icon StyledProps--main Text--icon StyledProps--padding-right-small"
          data-icon="ff-solid"
        >
          <svg
            fill="none"
            height="24"
            viewBox="0 0 32 32"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <lineargradient
              gradientUnits="userSpaceOnUse"
              id="ff-solid_svg__a"
              x1="6.837"
              x2="23.78"
              y1="2"
              y2="28.227"
            >
              <stop
                offset="0"
                stop-color="#ffc100"
              />
              <stop
                offset="1"
                stop-color="#e66b35"
              />
            </lineargradient>
            <g
              fill="url(#ff-solid_svg__a)"
            >
              <path
                d="M10.632 2.08A5.19 5.19 0 004.62 6.29L2.08 20.697a5.19 5.19 0 004.21 6.012l7.463 1.316a.2.2 0 00.208-.298L6.882 15.656c-.296-.469-.292-1.382.05-1.6.08-.05.313-.259.313-.259l.398-.436.066-.072c.37-.406.704-.773 1.76-1.302l.227-.079c1.548-.484 2.41-.449 3.27-.414.862.035 1.724.07 3.274-.414 1.272-.398 2.43-1.178 3.074-1.667a.622.622 0 01.9.15l4.596 7.362a.53.53 0 01-.101.696 8.694 8.694 0 01-2.834 1.538c-1.549.484-2.432.463-3.316.442-.884-.021-1.768-.042-3.318.442-1.314.41-1.868.948-2.07 1.204-.048.061-.07.21-.032.279.4.724 2.295 4.06 3.794 6.693a.998.998 0 00.695.49l3.07.541a5.19 5.19 0 006.012-4.21l2.54-14.408a5.19 5.19 0 00-4.21-6.012z"
              />
              <path
                d="M16.646 13.074c-1.303.34-2.093.316-2.882.29-.79-.024-1.58-.049-2.881.291-.744.216-1.178.592-1.587 1.072a.197.197 0 00-.018.23l2.482 4.069c.064.106.209.13.307.055.374-.283 1.28-.9 2.42-1.198 1.469-.383 2.348-.375 3.273-.366.717.007 1.462.014 2.53-.16a4.454 4.454 0 001.629-.635.347.347 0 00.094-.488L19.472 12.3a.421.421 0 00-.56-.128 9.57 9.57 0 01-2.265.902z"
              />
            </g>
          </svg>
        </span>
        common.purpose.cf.continuous common.plans.subscription
      </p>
      <div
        class="Layout--horizontal StyledProps--main"
      >
        <div
          class="Layout--horizontal StyledProps--main StyledProps--flex"
        >
          <div
            class="MultiStepProgressIndicator--dot MultiStepProgressIndicator--dotSuccess MultiStepProgressIndicator--spacing"
          />
        </div>
        <div
          class="Layout--horizontal StyledProps--main StyledProps--flex"
        >
          <div
            class="MultiStepProgressIndicator--statusBar"
          >
            <div
              class="MultiStepProgressIndicator--bar MultiStepProgressIndicator--fullBar"
            />
            <div
              class="MultiStepProgressIndicator--bar MultiStepProgressIndicator--barSuccess MultiStepProgressIndicator--fullBar"
            />
          </div>
          <div
            class="MultiStepProgressIndicator--dot MultiStepProgressIndicator--dotSuccess MultiStepProgressIndicator--spacing"
          />
        </div>
        <div
          class="Layout--horizontal StyledProps--main StyledProps--flex"
        >
          <div
            class="MultiStepProgressIndicator--statusBar"
          >
            <div
              class="MultiStepProgressIndicator--bar MultiStepProgressIndicator--fullBar"
            />
            <div
              class="MultiStepProgressIndicator--bar MultiStepProgressIndicator--barSuccess MultiStepProgressIndicator--fullBar"
            />
          </div>
          <div
            class="MultiStepProgressIndicator--dot MultiStepProgressIndicator--dotSuccess MultiStepProgressIndicator--spacing"
          />
        </div>
      </div>
      <p
        class="StyledProps--font StyledProps--main StyledProps--padding-top-large StyledProps--font-variation-h3"
      >
        authSettings.finalReview.step
      </p>
    </div>
    <div
      class="Layout--vertical StyledProps--main"
    >
      <div
        class="Layout--horizontal Layout--layout-spacing-small StyledProps--main"
      >
        <p
          class="StyledProps--font StyledProps--main StyledProps--font-variation-h5"
        >
          authSettings.costCalculator.newSubscription
        </p>
        <p
          class="StyledProps--font StyledProps--main newPlan StyledProps--font-weight-bold StyledProps--font-size-xsmall StyledProps--color StyledProps--color-white StyledProps--border"
        >
          ENTERPRISE
        </p>
      </div>
      <div
        class="Layout--vertical Layout--layout-spacing-large StyledProps--main"
      >
        <div
          class="Layout--horizontal StyledProps--main line StyledProps--flex StyledProps--flex-alignItems-start"
        >
          <p
            class="StyledProps--font StyledProps--main lineItem StyledProps--font-weight-semi-bold"
          >
            authSettings.costCalculator.developerLicenses
          </p>
          <div
            class="lineItem"
          >
             
            <p
              class="StyledProps--font StyledProps--main"
            >
              authSettings.unitPrice
              common.perMonth
            </p>
          </div>
          <p
            class="StyledProps--font StyledProps--main StyledProps--font-weight-semi-bold StyledProps--padding-right-xlarge"
          >
            $0.00
          </p>
        </div>
        <div
          class="Layout--horizontal StyledProps--main line StyledProps--flex StyledProps--flex-alignItems-start"
        >
          <p
            class="StyledProps--font StyledProps--main lineItem StyledProps--font-weight-semi-bold"
          >
            authSettings.costCalculator.maus
          </p>
          <div
            class="lineItem"
          >
             
            <div
              class="Layout--vertical StyledProps--main"
            >
              <p
                class="StyledProps--font StyledProps--main"
              >
                authSettings.unitPrice
                common.perMonth
              </p>
              <p
                class="StyledProps--font StyledProps--main"
              >
                authSettings.firstIncludedFree
              </p>
            </div>
          </div>
          <p
            class="StyledProps--font StyledProps--main StyledProps--font-weight-semi-bold StyledProps--padding-right-xlarge"
          >
            $0.00
          </p>
        </div>
        <div
          class="Layout--horizontal StyledProps--main line StyledProps--flex StyledProps--flex-alignItems-start"
        >
          <p
            class="StyledProps--font StyledProps--main lineItem StyledProps--font-weight-semi-bold"
          >
            authSettings.costCalculator.support
          </p>
          <div
            class="lineItem"
          >
             
            <div
              class="Layout--horizontal StyledProps--main"
            >
              <p
                class="StyledProps--font StyledProps--main"
              >
                authSettings.finalReview.premiumSupport
              </p>
              <p
                class="StyledProps--font StyledProps--main"
              >
                authSettings.finalReview.ofTotalSpend
              </p>
            </div>
          </div>
          <p
            class="StyledProps--font StyledProps--main StyledProps--font-weight-semi-bold StyledProps--padding-right-xlarge"
          >
            $0.00
          </p>
        </div>
        <div
          class="Layout--horizontal StyledProps--main line StyledProps--flex StyledProps--flex-alignItems-start"
        >
          <p
            class="StyledProps--font StyledProps--main lineItem StyledProps--font-weight-semi-bold"
          >
            authSettings.costCalculator.tax
          </p>
          <div
            class="lineItem"
          >
             
          </div>
          <p
            class="StyledProps--font StyledProps--main StyledProps--font-weight-semi-bold StyledProps--padding-right-xlarge"
          >
            $0.00
          </p>
        </div>
        <div
          class="Layout--horizontal StyledProps--main line StyledProps--flex StyledProps--flex-alignItems-start"
        >
          <p
            class="StyledProps--font StyledProps--main lineItem StyledProps--font-weight-semi-bold"
          >
            authSettings.yearlySubscriptionTotal
          </p>
          <div
            class="lineItem"
          >
             
            <p
              class="StyledProps--font StyledProps--main"
            >
              authSettings.costCalculator.autoRenew
            </p>
          </div>
          <p
            class="StyledProps--font StyledProps--main StyledProps--font-weight-semi-bold StyledProps--padding-right-xlarge"
          >
            $0.00
          </p>
        </div>
      </div>
      <div
        class="Layout--horizontal StyledProps--main payingToday StyledProps--flex StyledProps--flex-justifyContent-space-between"
      >
        <p
          class="StyledProps--font StyledProps--main StyledProps--font-variation-h4"
        >
          authSettings.costCalculator.payingToday
        </p>
        <p
          class="StyledProps--font StyledProps--main StyledProps--font-variation-h4"
        >
          $0.00
        </p>
      </div>
    </div>
    <div
      class="Layout--horizontal StyledProps--main footer"
    >
      <div
        class="Layout--horizontal Layout--layout-spacing-large StyledProps--main"
      >
        <button
          class="bp3-button Button--button StyledProps--font StyledProps--main Button--with-current-color Button--iconOnly Button--withLeftIcon Button--variation Button--variation-secondary"
          type="button"
        >
          <span
            class="bp3-icon bp3-icon-chevron-left StyledProps--main"
            icon="chevron-left"
          >
            <svg
              data-icon="chevron-left"
              height="16"
              viewBox="0 0 16 16"
              width="16"
            >
              <desc>
                chevron-left
              </desc>
              <path
                d="M7.41 8l3.29-3.29c.19-.18.3-.43.3-.71a1.003 1.003 0 00-1.71-.71l-4 4C5.11 7.47 5 7.72 5 8c0 .28.11.53.29.71l4 4a1.003 1.003 0 001.42-1.42L7.41 8z"
                fill-rule="evenodd"
              />
            </svg>
          </span>
          <span
            class="bp3-button-text"
          >
            back
          </span>
        </button>
        <button
          class="bp3-button Button--button StyledProps--font StyledProps--main Button--with-current-color Button--iconOnly Button--variation Button--variation-primary"
          type="button"
        >
          <span
            class="bp3-button-text"
          >
            authSettings.billing.subscribeNPay
          </span>
        </button>
      </div>
    </div>
  </div>
</div>
`;
